### PR TITLE
Test on chain sync

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: autonomy/chain/exceptions.py
Comment (angrybayblade) [inline autonomy/chain/exceptions.py:32]: Not relevant anymore

File: autonomy/chain/service.py
Comment (angrybayblade) [inline autonomy/chain/service.py:170]: Not relevant anymore

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (mariapiamo) [inline docs/advanced_reference/commands/autonomy_mint.md:138]: same suggestion of previous comment

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (mariapiamo) [inline docs/advanced_reference/commands/autonomy_mint.md:193]:
```suggestion
: Use the ERC20 token for securing autonomous services.
```

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (mariapiamo) [inline docs/advanced_reference/commands/autonomy_mint.md:196]: same suggestion of previous comment

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (mariapiamo) [inline docs/advanced_reference/commands/autonomy_mint.md:217]:
```suggestion
You can also use a custom ERC20 token as token for secure the autonomous service, to do this use `--token` flag to provide the address to the token of your choice:
```

File: docs/advanced_reference/commands/autonomy_service.md
Comment (mariapiamo) [inline docs/advanced_reference/commands/autonomy_service.md:38]: same comment above

File: docs/advanced_reference/commands/autonomy_service.md
Comment (mariapiamo) [inline docs/advanced_reference/commands/autonomy_service.md:34]: same comment above

File: docs/advanced_reference/commands/autonomy_service.md
Comment (mariapiamo) [inline docs/advanced_reference/commands/autonomy_service.md:138]: do we need to say somewhere that before using this an approval is required?

File: autonomy/chain/config.py
Comment (jmoreira-valory) [inline autonomy/chain/config.py:59]: Are these assignments correct?

File: autonomy/cli/helpers/chain.py
Comment (jmoreira-valory) [inline autonomy/cli/helpers/chain.py:376]: Why are these exceptions removed?

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_mint.md:22]:
```suggestion
: Use the custom-chain chain profile to interact with the Autonolas Protocol registry contracts. To define the custom-chain profile you must define some parameters and [contract addresses](../on_chain_addresses.md) as environment variables (see also the {{ autonolas_protocol }} documentation for more information):
```

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md:29]: The manager with the token util is only deployed on L1 chains eg ethereum and goerli and we provide these 2 configurations with the framework. Env vars are only used for L2 chains so no need for `SERVICE_MANAGER_TOKEN_ADDRESS `

File: docs/advanced_reference/commands/autonomy_service.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_service.md:138]: Approval will be taken care by the framework, the user just needs to have enough funds

File: autonomy/cli/helpers/chain.py
Comment (angrybayblade) [reply autonomy/cli/helpers/chain.py:376]: https://github.com/valory-xyz/open-autonomy/pull/2050#discussion_r1322411641 It's not required anymore since we're not going to use polling for checking if the component was minted or not

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (mariapiamo) [reply docs/advanced_reference/commands/autonomy_mint.md:29]: if this is only for L2s, why we have `CUSTOM_COMPONENT_REGISTRY_ADDRESS`, `CUSTOM_AGENT_REGISTRY_ADDRESS`,`CUSTOM_SERVICE_REGISTRY_TOKEN_UTILITY_ADDRESS` ?

File: autonomy/chain/config.py
Comment (angrybayblade) [reply autonomy/chain/config.py:59]: Fixed [378aebd](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/378aebd9c5e1015f5f2849b72d81a312c1e64778)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md:29]: Just in case if we deploy on some other L1 chain so the users won't have to wait until the next release

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md]: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md]: Addressed Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md]: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md]: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md]: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md:196]: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_mint.md]: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_service.md
Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_service.md]: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (mariapiamo) [reply docs/advanced_reference/commands/autonomy_mint.md:29]: Then **SERVICE_MANAGER_TOKEN_ADDRESS **  should be added as well. Agreed?

File: autonomy/chain/constants.py
Comment (mariapiamo) [inline autonomy/chain/constants.py:79]: it seems here that you call `service_manager` also on L1. Should we call this with correct name: service_manager_token?

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_mint.md:38]:
```suggestion
    !!! note

        For L2 chains you are only required to set

        - `CUSTOM_SERVICE_MANAGER_ADDRESS`,
        - `CUSTOM_SERVICE_REGISTRY_ADDRESS`,
        - `CUSTOM_GNOSIS_SAFE_PROXY_FACTORY_ADDRESS`,
        - `CUSTOM_GNOSIS_SAFE_SAME_ADDRESS_MULTISIG_ADDRESS` and
        - `CUSTOM_MULTISEND_ADDRESS`.
```

File: docs/advanced_reference/commands/autonomy_service.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:38]:
```suggestion
    !!! note

        For L2 chains you are only required to set

        - `CUSTOM_SERVICE_MANAGER_ADDRESS`,
        - `CUSTOM_SERVICE_REGISTRY_ADDRESS`,
        - `CUSTOM_GNOSIS_SAFE_PROXY_FACTORY_ADDRESS`,
        - `CUSTOM_GNOSIS_SAFE_SAME_ADDRESS_MULTISIG_ADDRESS` and
        - `CUSTOM_MULTISEND_ADDRESS`.
```

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (mariapiamo) [reply docs/advanced_reference/commands/autonomy_mint.md:29]: I see where this come from now. You call on both L1 and L2 service manager. Maybe for easier understanding would be better to explicitly show the difference

File: docs/advanced_reference/commands/autonomy_service.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:22]:
```suggestion
: Use the custom-chain profile to interact with the Autonolas Protocol registry contracts. This profile requires that you define some parameters and [contract addresses](../on_chain_addresses.md) as environment variables (see also the {{ autonolas_protocol }} documentation for more information):
```

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_mint.md:22]:
```suggestion
: Use the custom-chain profile to interact with the Autonolas Protocol registry contracts. This profile requires that you define some parameters and [contract addresses](../on_chain_addresses.md) as environment variables (see also the {{ autonolas_protocol }} documentation for more information):
```

File: autonomy/chain/constants.py
Comment (angrybayblade) [reply autonomy/chain/constants.py:79]: I'm storing both `service_manager` and `service_manager_token` as `service_manager` to avoid complications in the code

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_mint.md:90]:
```suggestion
: Update the already minted  protocol, contract, connection or skill with on-chain TOKEN_ID with the current package hash.
```

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_mint.md:197]:
```suggestion
: ERC20 token for securing the service.
```

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_mint.md:221]:
```suggestion
You can also use a custom ERC20 token as token to secure the service. Use the `--token` flag to provide the address of the token of your choice:
```

File: docs/advanced_reference/commands/autonomy_mint.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_mint.md:229]:
```suggestion
!!! warning "Important"

    If you have minted a service using a custom ERC20 token, then you have to use the same token to register the agent instances.
```

File: docs/advanced_reference/commands/autonomy_service.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:131]:
```suggestion
If an ERC20 token was used as bonding token you have to provide the same token address using `--token` flag:
```

File: docs/advanced_reference/commands/autonomy_service.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:137]:
```suggestion
Make sure your account holds enough funds for activating the service:
```

File: docs/advanced_reference/commands/autonomy_service.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:163]:
```suggestion
: ERC20 token to use as bond for the agent instances. You must specify the same token used when minitng the service. See the [`autonomy mint service`](./autonomy_mint.md#autonomy-mint-service) command.
```

File: docs/advanced_reference/commands/autonomy_service.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:184]:
```suggestion
If an ERC20 token was used as bonding token you have to provide the same token address using the `--token` flag:
```

File: docs/advanced_reference/commands/autonomy_service.md
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:190]:
```suggestion
Make sure your account holds enough funds to pay for the agent bonds.
```

**Review by jmoreira-valory — APPROVED** [review]
Please address submitted suggestions.

**Pull request merged**
